src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java

```diff
@@ -326,20 +343,92 @@ private void skipLineTerminator() {
         processLineTerminator(start, position());
     }
 
+    /**
+     * Scan the content of a string template expression.
+     *
+     * @param pos     start of literal
+     * @param endPos  start of embedded expression
+     */
+    private void scanEmbeddedExpression(int pos, int endPos) {
+        // If first embedded expression.
+        if (!isStringTemplate) {
+            checkSourceLevel(pos, Feature.STRING_TEMPLATES);
```

Comment:
If we check source level inside the `if`, we would only issue a preview error/warning once, which seems to be against JEP 12?

Reply:
The error is collated to the opening quote so one is sufficient. Embedded Expressions are not the feature. String templates is. The parser bails at that point.

```
>> cat Main.java
public class Main {
    public static void main(String... args) {
        int x = 10, y = 20;
        String s = STR."Adding \{x} and \{y} equals \{x + y}";
        String t = STR."Adding \{x} and \{y} equals \{x + y}";
    }
}
>> javac Main.java
Main.java:4: error: string templates are a preview feature and are disabled by default.
        String s = STR."Adding \{x} and \{y} equals \{x + y}";
                       ^
  (use --enable-preview to enable string templates)
1 error
```